parser.ts

```diff
@@ -219,8 +222,16 @@ export class Parser {
   private advance(): Token {
     this.previous = this.current;
     this.current = this.next;
-    this.line += this.nextWs + this.current.text;
+
+    if (this.line.lineNo.length && !this.line.rest.length) {
```

Comment:
This logic is suspect. But I'd expect the end result to concat the right chunks (`nextWs` and `this.current.text`) onto `this.line` in either case, just as before... I think this is OK.